source/fundamentals/data-formats/document-data-format-record.txt

```diff
@@ -225,4 +225,30 @@ execute a find operation on a collection of ``RecordTree`` types:
    .. output::
       :language: sh
 
-      RecordTree[content=Ranginui, left=RecordTree[content=..., left=RecordTree[content=..., right=RecordTree[content=Ikatere...]]
\ No newline at end of file
+      RecordTree[content=Ranginui, left=RecordTree[content=..., left=RecordTree[content=..., right=RecordTree[content=Ikatere...]]
+
+.. _fundamentals-records-get-codec:
+
+Retrieve the Record Codec
+-------------------------
+
+When working with records, consider using the ``RecordCodecProvider`` to get the
+record codec for Java records. For more information about codecs and why you might want
+to use them, see :ref:`Codecs <fundamentals-codecs>`. In summary, you use codecs
+to encode and decode Java objects to and from corresponding BSON types while
+minimizing duplicate code.
+
+.. tip::
+
+   When using the record codec, the annotations listed under
+   :ref:`Specify Component Conversion Using Annotations <fundamentals-records-annotations>`
+   are only supported on the record component.
+
+Be aware that you do not create a record codec directly. Instead,
+use the ``RecordCodecProvider`` to get the record codec. See the 
+`RecordCodecProvider API page <{+api+}/apidocs/bson-record-codec/org/bson/codecs/record/RecordCodecProvider.html>`__
+for more information.
+
+The record codec supports nested generic records. These are used the same way as
+other custom codecs as described on :ref:`Codecs <fundamentals-codecs>`. To see
+a complete example with custom codecs, see :ref:`Custom Codecs Example <codecs-custom-example>`.
```

Comment:
C: This section seems out of place in this section, as it has nothing to do with retrieving the record codec. Is there another place you can move this content?

Reply:
Moved this to the beginning where we touch briefly on the record codec since I think it fits there much better. Good call out.

Reply:
So after moving this, I didn't like how it flowed. This seemed like it was something that didn't fit or should be called out. I went to put it in the What's New topic since I thought that was where it would be best to go. On doing that, I discovered it was already there, so I just deleted this section.